lib/ace/executor.rb

```diff
@@ -6,13 +6,30 @@ def initialize(environment)
       @environment = environment
     end
 
-    def run_task(_targets, task, arguments, _options = {})
-      {
-        status: 'success',
-        result: {
-          output: "executed #{task.name}, with message `#{arguments['message']}` from #{@environment}"
-        }
-      }
+    def run_task(connection_info, task, arguments, _options = {})
+      if connection_info[:'remote-transport'] == 'panos' &&
+         task.files.first['filename'] == 'echo.sh'
+        [200, {
```

Comment:
These responses will also include a `"node"` key. See the example at the bottom of https://github.com/puppetlabs/bolt/blob/master/developer-docs/bolt-api-servers.md

Reply:
How do we identify a node with ACE? The `connection-info` can contain anything, based upon the schema. There are no identifiers.

Reply:
Would the node not be the certname/target for a device?

Reply:
> Would the node not be the certname/target for a device?

Not with a transport. We would need a flag set in the transport schema to set an attribute to act as an identifier. However, perhaps PE will set an identifier that can be passed to us along with the credentials?

Reply:
adding a 'node' with a fixed value to the mock endpoints because we will probably need to support it, but just don't at this stage.